apps/barista/src/components/live-example/live-example.ts

```diff
@@ -111,33 +132,89 @@ export class BaLiveExample implements OnDestroy {
    * @internal
    * The currently active tab for displaying a source (html, ts or scss).
    */
-  _activeTab: 'html' | 'ts' | 'scss';
+  _activeTab: BaSourceType;
+
+  /** @internal Whether the source code is hidden. */
+  _sourceHidden = false;
+
+  /** @internal Whether the source code is expanded to its max height. */
+  _isExpanded = false;
+
+  /** @internal Whether the source code has been copied. */
+  _copied$: BehaviorSubject<boolean> = new BehaviorSubject(false);
+
+  /** @internal Enhanced template source with line wrappers and code highlighting. */
+  _enhancedTemplateSource: string;
+
+  /** @internal Enhanced class source with line wrappers and code highlighting. */
+  _enhancedClassSource: string;
+
+  private _timerSubscription = Subscription.EMPTY;
 
   /** Whether the user has changed the active tab. */
   private _activeTabChanged = false;
 
+  /** Highlighter for highlight-ts syntax highlighting. */
+  private _highlighter: Highlighter<string>;
+
   constructor(
     private _componentFactoryResolver: ComponentFactoryResolver,
     private _injector: Injector,
     private _viewContainerRef: ViewContainerRef,
-  ) {}
+    private _platform: Platform,
+  ) {
+    registerLanguages(TypeScript, XML);
+    this._highlighter = init(htmlRender);
+  }
 
   ngOnDestroy(): void {
     if (this._componentRef) {
       this._componentRef.destroy();
     }
+    this._timerSubscription.unsubscribe();
   }
 
-  /** Whether one of the three sources has been set. */
+  /** @internal Whether one of the three sources has been set. */
   _hasSources(): boolean {
     return Boolean(this.classSource || this.templateSource);
   }
 
-  _setActiveTab(tab: 'html' | 'ts' | 'scss'): void {
+  /** @internal Sets the active tab. */
+  _setActiveTab(tab: BaSourceType): void {
     this._activeTab = tab;
     this._activeTabChanged = true;
   }
 
+  /** @internal Copies content of currently active tab to clipboard. */
+  _copyToClipboard(): void {
+    if (this._platform.isBrowser) {
+      let copyText;
+      switch (this._activeTab) {
+        case 'html':
+          copyText = this._templateSource;
+          break;
+        case 'ts':
+          copyText = this._classSource;
+          break;
+        case 'scss':
+          // TODO: copyText = this._scssSource;
+          break;
+      }
+
+      if (copyText) {
+        const textarea = createInputElement(copyText, 'textarea');
```

Comment:
We have now multiple copy to clipboard occurrences (headline link). Can we combine these copy to clipboard logic into a single function or service